doc/services/ipc/ipc_service/backends/ipc_service_icmsg.rst

```diff
@@ -0,0 +1,69 @@
+.. _ipc_service_backend_icmsg:
+
+ICMsg backend
+#############
+
+Inter core messaging backend (ICMsg) was created as a counter proposal to
+heavier RPMsg static vrings backend. It offers limitted features, but it small
+in size at the same time. It supports minimal functionalities to have the least
+possible memory footprint. ICMsg backend is build on top of :ref:`spsc_pbuf`.
+
+ICMsg backend supports registration of single endpoint on single instance. If
+application requires more than one communication channel, multiple instances
+must be defined. On each instance dedicated endpoint must be created.
+
+Bonding
+*******
+
+When the endpoint is registered, a magic number is written to the shared memory
+and a signal is sent to other domain/CPU. After that attempt to read the magic
+number from the other side of the communication is done. If the magic number is
+already there it means that the other communication side has already registered
+endpoint on its side and bound callback is called. If there is no magic number
+yet, the endpoint waits to receive signal from the other domain/CPU to read the
+magic number.
```

Comment:
```suggestion
When the endpoint is registered, the following happens on each one
of the domains (or CPUs) connected through the IPC instance:

1. The domain (or CPU) writes a magic number to its ``TX-region`` of the shared memory.
#. It then sends a signal to the other domain or CPU.
#. The receiving domain or CPU attempts to read the magic number from its ``RX-region``
   (where ``TX-region`` for an endpoint is ``RX-region`` for the other communication
   side, and viceversa).

If the magic number is found in the ``RX-region`` of the shared memory for a specific
domain or CPU, it means that the other communication side has already registered the
endpoint on its side, and the first endpoint calls a bound callback. If there is no
magic number yet, the endpoint waits to receive the signal from the other domain or
CPU to read the magic number.
```

@emob-nordic Please check this interpretation

Reply:
@FrancescoSer i made some changes in this part of the documentation. Please have a look. Hopefully it is clearer now.